src/v/archival/ntp_archiver_service.h

```diff
@@ -289,6 +289,11 @@ class ntp_archiver {
       std::optional<std::reference_wrapper<retry_chain_node>> source_rtc
       = std::nullopt);
 
+    /// Waits for this node to become leader, syncing archival metadata within
+    /// the current term. Returns false if the sync was not successful.
+    /// Must be called before updating archival metadata.
+    ss::future<bool> sync_for_tests();
```

Comment:
You can use a 'friend' trick to makes private state available to the test. Something like:

```cpp
// foo.cc
class test_accessor; // not implemented in foo.cc

class Foo {
  friend class test_accessor;
...
 void sync(); // private method
};

// test.cc
class test_accessor {
public:
    void call_sync(Foo& foo) {
       foo.sync();
    }
}

BOOST_TEST(test_sync) {
    Foo foo;
    test_accessor a;
   a.sync(foo);
   ...
}
```

Reply:
Hrm, not a bad idea, though I think i'm going to punt for now. I'm hoping this is clear enough to not be misused.